support/controlplane-component/controlplane-component.go

```diff
@@ -0,0 +1,339 @@
+package controlplanecomponent
```

Comment:
we'll want to follow up for all this core logic to be unit tested here as we move forward even though is implicitly tested by the components.